socket/index.js

```diff
@@ -73,10 +73,94 @@ io.on('connection', (socket) => {
 		} else {
 			allUsers[user.sessionId] = curr;   
 		}
+
+		socket.join(user.sessionId);
 		// Update current player
 		socket.emit('user-joined', allUsers[user.sessionId]);
 		// Update all players
-		socket.broadcast.emit('user-joined', allUsers[user.sessionId]);
+		socket.broadcast.to(user.sessionId).emit('user-joined', allUsers[user.sessionId]);
+		// Check if any cards were selected before user joined
+		socket.emit('watch-submit-card', allCards[user.sessionId]);
+	})
+
+	// Select card event
+	socket.on('set-card', (data) => {
+		let curr = {}
+
+		Session.findOne( { id: data.session }, (err, session) => {
+			if (err) console.error(err);
+			
+			// If cards have been flipped, prohibit changing card selection
+			if (session.state != true) {
+
+				User.find({}, function (err, users) {
+					if (err) console.error();
+			
+					i = userExists(users, data.user);
+					users[i].card = data.card;
+					session.users[userExists(session.users, data.user)].card = data.card;
+					
+					session.save((err) => {
+					if (err) console.error(`Error updating user role in session: ${err}`);
+					}) 
+					users[i].save((err) => {
+					if (err) console.error(`Error updating user role: ${err}`);
+					})
+
+					// if user de-selects card
+					if (data.card == null) {
+						currSession = allCards[data.session]
+						delete currSession[data.user]
+					} else {
+						curr[data.user] = {card: data.card}
+
+						if (data.session in allCards) {
+							_.merge(allCards[data.session], curr);
+						} else {
+							allCards[data.session] = curr;   
+						}
+					}
+		
+					socket.emit('watch-submit-card', allCards[data.session]);
+					socket.broadcast.to(data.session).emit('watch-submit-card', allCards[data.session]);
+				})
+			}
+		})
+		
+	})
+
+	socket.on('set-cards-up', (sessionId) => {
+		Session.findOne( { id: sessionId }, (err, session) => {
+			if (err) console.error(err);
+
+			if (session.state == null) {
+				session.state = true
+			} else if (session.state == false) {
+				session.state = true
```

Comment:
Can't you just do `session.state = !session.state`?

Reply:
I don't think that would work, because there are three states that it can be (null, false, true)